azure/scope/managedcontrolplane.go

```diff
@@ -472,7 +472,7 @@ func (s *ManagedControlPlaneScope) ManagedClusterSpec() (azure.ManagedClusterSpe
 }
 
 // GetAgentPoolSpecs gets a slice of azure.AgentPoolSpec for the list of agent pools.
```

Comment:
the above comment needs to change to reflect the new method name if we're going to keep that, and ditto the `"scope.ManagedControlPlaneScope.GetAgentPoolSpecs"` string reference in the logger needs to be changed to `"scope.ManagedControlPlaneScope.GetAllAgentPoolSpecs"`

Reply:
good catch! thanks

Reply:
removed the lgtm to allow you to update this but otherwise lgtm

Reply:
updated! also rebased and squashed the commits